pkg/agent/run.go

```diff
@@ -77,11 +78,22 @@ func Run(ctx context.Context, cfg cmds.Agent) error {
 	}
 
 	cfg.DataDir = filepath.Join(cfg.DataDir, "agent")
+	os.MkdirAll(cfg.DataDir, 0700)
 
 	if cfg.ClusterSecret != "" {
 		cfg.Token = "K10node:" + cfg.ClusterSecret
 	}
 
+	lb, err := loadbalancer.Setup(ctx, cfg)
+	if err != nil {
+		return err
+	}
+	if lb != nil {
+		if cfg.ServerURL, err = lb.LoadBalancerServerURL(); err != nil {
```

Comment:
doesn't this setup cause tls CN mismatch?

Reply:
If I understand the question correctly it should not cause an issue because the load balancer is running on 127.0.0.1, and clients connecting through the load balancer will use that IP for verifying certs, and since the signed cert to verify (returned from the servers) contains ip 127.0.0.1 in the sans of the cert the client will accept the connection.

Reply:
you understood my question correctly. :-)
thank you for explanation.